tests/thumbnail_test_provider.py

```
@@ -0,0 +1,54 @@
+from rero_invenio_thumbnails.contrib.api import BaseProvider
+
+
+class TestThumbnailProvider(BaseProvider):
+    """Deterministic in-memory thumbnail provider for the test suite.
+
+    Uses a class-level ``covers`` dict so that all instances created by
+    ``get_thumbnail_url`` share the same mapping without requiring injection
+    into individual instances.
+    """
+
+    covers = {}
```

Ruff (0.15.6)

[warning] Mutable default value for class attribute

(RUF012)